homeassistant/components/maxcube/climate.py

```diff
@@ -199,49 +172,30 @@ def hvac_action(self):
     @property
     def target_temperature(self):
         """Return the temperature we try to reach."""
-        device = self._cubehandle.cube.device_by_rf(self._rf_address)
-        if (
-            device.target_temperature is None
-            or device.target_temperature < self.min_temp
-            or device.target_temperature > self.max_temp
-        ):
+        temp = self._device.target_temperature
+        if temp is None or temp < self.min_temp or temp > self.max_temp:
             return None
-        return device.target_temperature
+        return temp
 
     def set_temperature(self, **kwargs):
         """Set new target temperatures."""
-        if kwargs.get(ATTR_TEMPERATURE) is None:
-            return False
-
-        target_temperature = kwargs.get(ATTR_TEMPERATURE)
-        device = self._cubehandle.cube.device_by_rf(self._rf_address)
-
-        cube = self._cubehandle.cube
-
-        with self._cubehandle.mutex:
-            try:
-                cube.set_target_temperature(device, target_temperature)
-            except (socket.timeout, OSError):
-                _LOGGER.error("Setting target temperature failed")
-                return False
+        self._set_target(None, kwargs.get(ATTR_TEMPERATURE))
```

Comment:
The temperature parameter may be `None` so we should guard against that before calling this method. We usually raise a `ValueError` with appropriate message as exception argument on invalid service input.

Reply:
In that case, it will call cube.set_temperature_mode(None, None), which does nothing. But I will change it to raise an exception.